TRAMS-API/Mapping/GetProjectAcademiesResponseMapper.cs

```diff
@@ -0,0 +1,60 @@
+﻿using API.Models.Downstream.D365;
+using API.Models.Upstream.Enums;
+using API.Models.Upstream.Response;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace API.Mapping
+{
+    public class GetProjectAcademiesResponseMapper : IMapper<AcademyTransfersProjectAcademy,
+                                                             Models.Upstream.Response.GetProjectsAcademyResponseModel>
+    {
+        public GetProjectsAcademyResponseModel Map(AcademyTransfersProjectAcademy academy)
+        {
+            return new GetProjectsAcademyResponseModel
+            {
+                ProjectAcademyId = academy.AcademyTransfersProjectAcademyId,
+                AcademyId = academy.AcademyId,
+                AcademyName = academy.AcademyName,
+                ProjectId = academy.ProjectId,
+                EsfaInterventionReasons = ExtractEsfaInterventionReasons(academy),
+                EsfaInterventionReasonsExplained = academy.EsfaInterventionReasonsExplained,
+                RddOrRscInterventionReasons = ExtractRddorRscInterventionReason(academy),
+                RddOrRscInterventionReasonsExplained = academy.RddOrRscInterventionReasonsExplained,
+                AcademyTrusts = academy.ProjectAcademyTrusts == null || academy.ProjectAcademyTrusts.Count == 0
+                                ? new List<GetAcademyTrustsResponseModel>()
+                                : academy.ProjectAcademyTrusts.Select(t => new GetAcademyTrustsResponseModel
+                                {
+                                    ProjectTrustId = t.ProjectAcademyTrustId,
+                                    TrustId = t.TrustId,
+                                    TrustName = t.TrustName
+                                })
+                                                        .ToList()
+            };
+        }
+
+        private static List<RddOrRscInterventionReasonEnum> ExtractRddorRscInterventionReason(AcademyTransfersProjectAcademy a)
+        {
+            return string.IsNullOrEmpty(a.RddOrRscInterventionReasons)
+                   ? new List<RddOrRscInterventionReasonEnum>()
+                   : a.RddOrRscInterventionReasons?
+                      .Split(',')
+                      .Select(v => Enum.Parse<Models.D365.Enums.RddOrRscInterventionReasonEnum>(v))
+                      .Select(v => MappingDictionaries.RddOrRscInterventionReasonEnumMap.Where(d => d.Value == v).FirstOrDefault().Key)
```

Comment:
Warning: Unlikely to be the case but FirstOrDefault() could yield a null value if no matches found.  I suspect this will be covered off by validation of the input model.

Reply:
Will need to think about this. This mapper converts from D365 to response so there's no validation of the input. Problem could arise if D365 uses an unknown enum value.